pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -586,8 +621,21 @@ public SegmentCompletionProtocol.Response extendBuildTime(final String instanceI
           case COMMITTER_DECIDED:
             return fail(instanceId, offset);
           case COMMITTER_NOTIFIED:
-            return committerNotifiedExtendBuildTime(instanceId, offset, extTimeSec, now);
+            if (!_pauselessConsumptionEnabled) {
+              return committerNotifiedExtendBuildTime(instanceId, offset, extTimeSec, now);
```

Comment:
Change in the commit protocol changes the state in which we get the `extendBuildTime` request